plugin/net/cni.go

```diff
@@ -183,16 +182,21 @@ func assignBridgeIP(bridgeName string, ipnet net.IPNet) error {
 	return nil
 }
 
+// As of CNI 0.5 spec:
+//   "Plugins should generally complete a DEL action without error even if some resources are missing"
+// this method should therefore return nil in most, if not all, cases.
 func (c *CNIPlugin) CmdDel(args *skel.CmdArgs) error {
 	conf, err := loadNetConf(args.StdinData)
 	if err != nil {
-		return err
+		logOnStderr(err)
```

Comment:
I'm not 100% convinced about this change; if we can't even load the config that's pretty bad.  And returning immediately without trying to do anything is not really in the spirit of "complete a DEL action"

However I think we should go ahead because Kubernetes is being super-dumb about sending us rubbish and then getting stuck when we complain, so let's try never erroring and see how it goes.